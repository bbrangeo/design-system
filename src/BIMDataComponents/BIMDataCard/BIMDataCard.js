import BIMDataCard from './BIMDataCard.vue';

export default BIMDataCard;