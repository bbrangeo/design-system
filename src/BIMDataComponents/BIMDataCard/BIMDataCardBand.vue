<template>
  <div class="bimdata-card__brand">
    <div class="bimdata-card__brand--left">
      <slot name="left"></slot>
      <span v-if="text">{{text}}</span>
    </div>
    <slot name="right"></slot>
  </div>
</template>

<script>
import BIMDataBurgerMenu from "../BIMDataBurgerMenu/BIMDataBurgerMenu.vue"
export default {
  components: {
    BIMDataBurgerMenu
  },
  props: {
    text: {
      type: String
    }
  }
}
</script>

<style lang="scss">
  // import BIMDATA VARIABLES
  @import "../../assets/scss/_BIMDataVariables.scss";

  // import BIMDATA STYLE COMPONENT
  @import "./_BIMDataCardBrand.scss";
</style>