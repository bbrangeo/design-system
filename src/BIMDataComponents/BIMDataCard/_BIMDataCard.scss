.bimdata-card{
  background-color: $color-white;
  box-shadow: 0px 2px 10px rgba($color-black, .07);
  &__content, &__footer{
    font-size: calculateEm(12px);
  }
  &__brand{
    justify-content: space-between;
    &--left{
      margin-right: $spacing-unit;
      display: flex;
      align-items: center;
      svg{
        margin-right: calc(#{$spacing-unit} / 2);
      }
    }
  }
  &__header{
    padding: 10px $spacing-unit;
    height: auto;
    display: flex;
    align-items: center;
    justify-content: space-between;
    background-color: $color-tertiary-lightest;
    color: $color-primary;
    font-size: calculateEm(13px);
  }
  &__submenu{
    padding: calc(#{$spacing-unit} / 2) $spacing-unit;
    display: flex;
    align-items: center;
    position: relative;
    background: $color-white;
    box-shadow: 0px 2px 4px rgba($color-black, .1);
    font-size: calculateEm(12px);
  }
  &__content{
    padding: $spacing-unit;
    color: $color-tertiary-dark;
  }
  &__footer{
    padding: 0 $spacing-unit $spacing-unit;
  }
}