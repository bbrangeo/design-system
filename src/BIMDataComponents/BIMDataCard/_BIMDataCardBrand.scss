.bimdata-card__brand {
  justify-content: space-between;
  &--left {
    margin-right: $spacing-unit;
    display: flex;
    align-items: center;
    svg {
      margin-right: calc(#{$spacing-unit} / 2);
    }
  }
}
