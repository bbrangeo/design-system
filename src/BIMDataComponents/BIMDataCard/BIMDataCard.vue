<template>
  <div class="bimdata-card" :style="{ 'min-width': width}">
    <BIMDataCardBand :text="titleHeader" class="bimdata-card__header" v-if="titleHeader || $scopedSlots.headerIcons">
      <template #right>
        <slot name="headerIcons"></slot>
      </template>
    </BIMDataCardBand>
    <BIMDataCardBand class="bimdata-card__submenu" :text="submenuText" v-if="$scopedSlots.left || submenuText || $scopedSlots.right">
      <template #left>
        <slot name="left"></slot>
      </template>
      <template #right>
        <slot name="right"></slot>
      </template>
    </BIMDataCardBand>
    <main class="bimdata-card__content" v-if="$scopedSlots.content">
      <slot name="content"></slot>
    </main>
    <footer class="bimdata-card__footer" v-if="$scopedSlots.footer">
      <slot name="footer"></slot>
    </footer>
  </div>
</template>

<script>
import BIMDataCardBand from "../BIMDataCard/BIMDataCardBand.vue"

export default {
  components: {
    BIMDataCardBand
  },
  props: {
    titleHeader: {
      type: String
    },
    submenuText: {
      type: String
    },
    width: {
      type: [Number, String],
      default: "215px"
    }
  }
}
</script>

<style lang="scss">
  // import BIMDATA VARIABLES
  @import "../../assets/scss/_BIMDataVariables.scss";

  // import BIMDATA MIXINS
  @import "../../assets/scss/mixins/_font-size.scss";

  // import BIMDATA STYLE COMPONENT
  @import "./_BIMDataCard.scss";
</style>