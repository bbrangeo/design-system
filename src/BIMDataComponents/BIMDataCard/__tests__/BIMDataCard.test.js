import { shallowMount } from "@vue/test-utils";
import BIMDataCard from "../BIMDataCard.vue";

describe("BIMDataCard", () => {
  it("should render component and match snapshot", () => {
    const wrapper = shallowMount(BIMDataCard);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render component with title header", () => {
    const wrapper = shallowMount(BIMDataCard, {
      propsData: {
        titleHeader: "Hearder title card",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render component with submenu text", () => {
    const wrapper = shallowMount(BIMDataCard, {
      propsData: {
        submenuText: "Submenu text card",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render component with a custom width", () => {
    const wrapper = shallowMount(BIMDataCard, {
      propsData: {
        width: "350px",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render a card with a headerIcons slot and match snapshot", () => {
    const wrapper = shallowMount(BIMDataCard, {
      slots: {
        headerIcons: "<span>header icons slot</span>",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render a card with a left slot and match snapshot", () => {
    const wrapper = shallowMount(BIMDataCard, {
      slots: {
        left: "<span>left slot</span>",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render a card with a left slot and match snapshot", () => {
    const wrapper = shallowMount(BIMDataCard, {
      slots: {
        right: "<span>right slot</span>",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render a card with a content slot and match snapshot", () => {
    const wrapper = shallowMount(BIMDataCard, {
      slots: {
        content: "<span>content slot</span>",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render a card with a footer slot and match snapshot", () => {
    const wrapper = shallowMount(BIMDataCard, {
      slots: {
        footer: "<span>footer slot</span>",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
