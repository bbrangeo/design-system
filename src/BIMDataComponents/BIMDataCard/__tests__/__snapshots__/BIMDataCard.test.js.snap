// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BIMDataCard should render a card with a content slot and match snapshot 1`] = `
"<div class=\\"bimdata-card bimdata-card__default\\" style=\\"min-width: 215px; border-radius: 0px;\\">
  <!---->
  <!---->
  <div class=\\"bimdata-card__content\\"><span>content slot</span></div>
  <!---->
</div>"
`;

exports[`BIMDataCard should render a card with a footer slot and match snapshot 1`] = `
"<div class=\\"bimdata-card bimdata-card__default\\" style=\\"min-width: 215px; border-radius: 0px;\\">
  <!---->
  <!---->
  <!---->
  <footer class=\\"bimdata-card__footer\\"><span>footer slot</span></footer>
</div>"
`;

exports[`BIMDataCard should render a card with a headerIcons slot and match snapshot 1`] = `
"<div class=\\"bimdata-card bimdata-card__default\\" style=\\"min-width: 215px; border-radius: 0px;\\">
  <bimdatacardband-stub class=\\"bimdata-card__header\\"></bimdatacardband-stub>
  <!---->
  <!---->
  <!---->
</div>"
`;

exports[`BIMDataCard should render a card with a left slot and match snapshot 1`] = `
"<div class=\\"bimdata-card bimdata-card__default\\" style=\\"min-width: 215px; border-radius: 0px;\\">
  <!---->
  <bimdatacardband-stub class=\\"bimdata-card__submenu\\"></bimdatacardband-stub>
  <!---->
  <!---->
</div>"
`;

exports[`BIMDataCard should render a card with a left slot and match snapshot 2`] = `
"<div class=\\"bimdata-card bimdata-card__default\\" style=\\"min-width: 215px; border-radius: 0px;\\">
  <!---->
  <bimdatacardband-stub class=\\"bimdata-card__submenu\\"></bimdatacardband-stub>
  <!---->
  <!---->
</div>"
`;

exports[`BIMDataCard should render component and match snapshot 1`] = `
"<div class=\\"bimdata-card bimdata-card__default\\" style=\\"min-width: 215px; border-radius: 0px;\\">
  <!---->
  <!---->
  <!---->
  <!---->
</div>"
`;

exports[`BIMDataCard should render component with a custom width 1`] = `
"<div class=\\"bimdata-card bimdata-card__default\\" style=\\"min-width: 350px; border-radius: 0px;\\">
  <!---->
  <!---->
  <!---->
  <!---->
</div>"
`;

exports[`BIMDataCard should render component with submenu text 1`] = `
"<div class=\\"bimdata-card bimdata-card__default\\" style=\\"min-width: 215px; border-radius: 0px;\\">
  <!---->
  <bimdatacardband-stub text=\\"Submenu text card\\" class=\\"bimdata-card__submenu\\"></bimdatacardband-stub>
  <!---->
  <!---->
</div>"
`;

exports[`BIMDataCard should render component with title header 1`] = `
"<div class=\\"bimdata-card bimdata-card__default\\" style=\\"min-width: 215px; border-radius: 0px;\\">
  <bimdatacardband-stub text=\\"Hearder title card\\" class=\\"bimdata-card__header\\"></bimdatacardband-stub>
  <!---->
  <!---->
  <!---->
</div>"
`;
